src/main/java/com/tfgm/controllers/TramController.java

```diff
@@ -115,35 +115,35 @@ public void update() throws URISyntaxException, IOException {
         for (int j = 0; j < 4; j++) {
           if (!(currentStation.getString("Dest" + j).equals(""))) {
             if (currentStation.getString("Status" + j).equals("Departing")) {
-              NewTramStop newTramStop =
+              TramStop tramStop =
                   tramStopHashMap.get(
                       currentStation.getString("StationLocation").replaceAll("[^A-Za-z]+", "")
                           + currentStation.getString("Direction"));
 
-              if (newTramStop != null) {
-                if (!newTramStop.getLastUpdated().contains((getUpdateString(currentStation, j)))) {
-                  newTramStop.addToLastUpdated(getUpdateString(currentStation, j));
+              if (tramStop != null) {
```

Comment:
When moving your logic into the service layer, try to use more methods and variables with meaningful names to make sure the code is readable. At the moment it is quite difficult to decipher what is going on in the code.